hello-signing.md

```diff
@@ -0,0 +1,129 @@
+
+## Scenarios
+
+To demonstrate how to store and sign a set of supply chain artifacts, we'll walk through a set of scenarios:
+- Singing a single container image
+- Publishing the container image across public registries, 
+- Importing public content to a private registry
+- Promoting from dev through production
+
+![](../media/notary-e2e-scenarios.svg)
+
+- Wabbit Networks is a small software company that produces network monitoring software.
+- ACME Rockets wishes to acquire network monitoring software.
+- ACME Rockets doesn't know about Wabbit Networks, but finds their [net-monitor software in Docker Hub](https://hub.docker.com/r/wabbitnetworks/net-monitor)
+- Since they've never heard of Wabbit Networks, they're a little reluctant to run network software without some validations.
+- ACME Rockets has a policy to only import Docker Hub certified software, or other approved vendors.
+- Wabbit Networks works with Docker Hub to get certified, to help with their customer confidence.
+- ACME Rockets will only deploy software that's been scanned and approved by the ACME Rockets security team. They know it's been approved because all approved software has been signed by the ACME Rockets security team.
+
+## Getting Started
+- Setup a few environment variables.  
+  >Note see [Simulating a Registry DNS Name](#simulating-a-registry-dns-name) to use `registry.wabbit-networks.io`
+  ```bash
+  export PORT=5000
+  export REGISTRY=localhost:${PORT}
+  export REPO=${REGISTRY}/net-monitor
+  export IMAGE=${REPO}:v1
+  ```
+- Install [Docker Desktop](https://www.docker.com/products/docker-desktop) for local docker operations
+- Run a local instance of the [CNCF Distribution Registry][cncf-distribution]
+  ```bash
+  docker run -d -p ${PORT}:5000 ghcr.io/oras-project/registry:latest
+  ```
+- Acquire the Notation CLI  
+Notation releases can be found at: [Notation Releases][notation-releases]  
+  ```bash
+  curl -O ~/bin/notation https://github.com/notaryproject/notation/releases/download/v0.0.1-alpha/notation-v0.0.1-alpha.1-linux-amd64
+  curl -O ~/.docker/cli-plugins/docker-notation https://github.com/notaryproject/notation/releases/download/v0.0.1-alpha/docker-notation-v0.0.1-alpha.1-linux-amd64
+  ```
+
+## Building and Pushing
+- Build and Push the `net-monitor` software
+  ```bash
+  docker build -t $IMAGE https://github.com/wabbit-networks/net-monitor.git#main
+
+  docker push $IMAGE
+  ```
+- List the image, and any associated signatures
+  ```bash
+  notation list $IMAGE
+  ```
+  At this point, the results are empty, as there are no existing signatures
+
+## Signing a Container Image
+
+As you automate the signing of content, you'll most likely want to create and store the private keys in a key vault. Detailed production steps will be covered later. To get things started quickly, the notation cli supports generating self signed key-pairs.
+- Generating a Private Key  
+  The following will generate an X-509 private key under the ./notary/ directory
```

Comment:
It would be useful to write this line as a comment instead so it's easier to read

  ```bash
# The following will generate an X-509 private key under the ./notary/ directory
  notation cert generate "wabbit-networks.io"
  ```
Also, if someone is copy pasting from here, they get free documentation.

Reply:
To keep the code to a minimum, I'm working to keep the text outside the copy/paste commands.